### DISCUSSION TITLE
`git log` output incorrect only in action

### DISCUSSION THREAD

**Discussion by ero-qt** [community/community#68813] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi,

I'm trying to create a changelog via `git log` and its formatting. I fetch the tag of the previous release using [`cardinalby/git-get-release-action@v1`](https://github.com/marketplace/actions/git-get-release-action). In a `run` step, I then first call `git fetch --all --tags`, and place the following into the output:
```bash
{
  echo 'changelog<<EOF'
  git log --pretty=format:"- %s (%H)" ${{ steps.previous-release.outputs.tag_name }}..HEAD
  echo EOF
} >> $GITHUB_OUTPUT
```
Despite there being multiple commits between the previous release's tag and `HEAD`, only 1 line is shown. I can confirm this by running the same command locally, which shows all commits between these two targets.

What causes this? How can I fix it?

**Comment by vishalkrishna8** [comment]
Heyy @just-ero,
Use the script below:
`  - name: Create Changelog
  id: changelog
  run: |
    echo 'changelog<<EOF' >> $GITHUB_ENV
    git log --pretty=format:"- %s (%H)" ${{ steps.previous-release.outputs.tag_name }}..HEAD >> $GITHUB_ENV
    echo 'EOF' >> $GITHUB_ENV
   `
1.This stores the changelog in an environment variable for later use.

2.Adjust variable usage as needed in your workflow.

This ensures you capture the entire changelog in your GitHub Actions workflow.

**Reply by ero-qt** [reply]
This is the exact same as what I did before. Additionally, this doesn't work.  
```bash
git fetch --all --tags
echo 'changelog<<EOF' >> $GITHUB_OUTPUT
git log --pretty=format:"- %s (%H)" ${{ steps.previous-release.outputs.tag_name }}..HEAD >> $GITHUB_OUTPUT
echo 'EOF' >> $GITHUB_OUTPUT
```
```
Error: Unable to process file command 'output' successfully.
Error: Invalid value. Matching delimiter not found 'EOF'
```

Please actually test whatever help you provide instead of (I assume) blindly copying something from some AI assistant.

**Reply by vishalkrishna8** [reply]
When you don't know anything don't assume anything, As for your question is concern you must have made some silly mistakes or syntax error. 
And like every time, this time also do not mark your questions by answering them yourself.

**Reply by ero-qt** [reply]
I mean I provided you with the code I used. You can't say that *I* have a syntax error when the code is an exact copy of yours.

And what's wrong with answering my own questions once I find a solution to the exact problem I was having? Just seems like good practice (even encouraged in most other forums).

**Reply by vishalkrishna8** [reply]
Syntax error meant review the entire script for any syntax errors or missing quotes that might cause issues.

**Reply by ero-qt** [reply]
The job runs fine. It wouldn't be able to run if there were a mistake in the syntax of the YAML file.

**Reply by vishalkrishna8** [reply]
If there is debugging command, you can check to get more information about where the problem occurs.

**Comment by SaifurRR** [comment]
can you capture entire change log?

**Comment by ero-qt** [comment, marked as answer]
The issue was with the fetch depth of `actions/checkout@v4`. Changing it to
```yml
- uses: actions/checkout@v4
  with:
    fetch-depth: 0
    fetch-tags: true
```
fixed everything.
